airflow/contrib/hooks/mongo_hook.py

```diff
@@ -0,0 +1,112 @@
+    You can specify connection string options in extra field of your connection
+    https://docs.mongodb.com/manual/reference/connection-string/index.html#connection-string-options
+    ex.
+        {replicaSet: test, ssl: True, connectTimeoutMS: 30000}
+    """
+    conn_type = 'MongoDb'
+
+    def __init__(self, conn_id='mongo_default', *args, **kwargs):
+        super(MongoHook, self).__init__(source='mongo')
+
+        self.mongo_conn_id = conn_id
+        self.connection = self.get_connection(conn_id)
+        self.extras = self.connection.extra_dejson
+
+    def get_conn(self):
+        """
+        Fetches PyMongo Client
+        """
+        conn = self.connection
+
+        uri = 'mongodb://{creds}{host}{port}/{database}'.format(
+            creds='{}:{}@'.format(
+                conn.login, conn.password
+            ) if conn.login is not None else '',
+
+            host=conn.host,
+            port='' if conn.port is None else ':{}'.format(conn.port),
+            database='' if conn.schema is None else conn.schema
+        )
+
+        # Mongo Connection Options dict that is unpacked when passed to MongoClient
+        options = self.extras
+
+        # If we are using SSL disable requiring certs from specific hostname
+        if options.get('ssl', False):
+            options.update({'ssl_cert_reqs': CERT_NONE})
+
+        return MongoClient(uri, **options)
+
+    def get_collection(self, mongo_collection, mongo_db=None):
+        """
+        Fetches a mongo collection object for querying.
+
+        Uses connection schema as DB unless specified
+        """
+        mongo_db = mongo_db if mongo_db is not None else self.connection.schema
+        mongo_conn = self.get_conn()
+
+        return mongo_conn.get_database(mongo_db).get_collection(mongo_collection)
+
+    def aggregate(self, mongo_collection, aggregate_query, mongo_db=None, **kwargs):
+        """
+        Runs and aggregation pipeline and returns the results
+        https://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.aggregate
+        http://api.mongodb.com/python/current/examples/aggregation.html
+        """
+        collection = self.get_collection(mongo_collection, mongo_db=mongo_db)
+
+        return collection.aggregate(aggregate_query, **kwargs)
+
+    def find(self, mongo_collection, query, find_one=False, mongo_db=None, **kwargs):
+        """
+        Runs a mongo find query and returns the results
+        https://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.find
+        """
+        collection = self.get_collection(mongo_collection, mongo_db=mongo_db)
+
+        if find_one:
+            return collection.find_one(query, **kwargs)
+        else:
+            return collection.find(query, **kwargs)
+
+    def insert_one(self, mongo_collection, doc, mongo_db=None, **kwargs):
+        """
+        Inserts a single document into a mongo collection
```

Comment:
The other methods link to their underlying docs, probably should here, too.

https://api.mongodb.com/python/current/api/pymongo/collection.html#pymongo.collection.Collection.insert_one